proto/vtctldata.proto

```diff
@@ -1303,6 +1359,68 @@ message ValidateVSchemaResponse {
   map<string, ValidateShardResponse> results_by_shard = 2;
 }
 
+message WorkflowDeleteRequest {
+  string keyspace = 1;
+  string workflow = 2;
+  bool keep_data = 3;
+  bool keep_routing_rules = 4;
+}
+
+message WorkflowDeleteResponse {
+  message TabletInfo {
+    string tablet = 1;
+    // The value is based on the query result's RowsAffected
+    // being 0 or not.
+    bool deleted = 2;
+  }
+  string summary = 1;
+  repeated TabletInfo details = 2;
+}
+
+message WorkflowStatusRequest {
+  string keyspace = 1;
+  string workflow = 2;
+}
+
+message WorkflowStatusResponse {
+  message TableCopyState {
+    int64 rows_copied = 1;
+    int64 rows_total = 2;
+    int32 rows_percentage = 3;
+    int64 bytes_copied = 4;
+    int64 bytes_total = 5;
+    int32 bytes_percentage = 6;
+  }
+  message ShardStreamState {
+    int32 id = 1;
+    string tablet = 2;
+    string source_shard = 3;
+    string position = 4;
+    string status = 5;
+    string info = 6;
+  }
+  message ShardStreams {
+    repeated ShardStreamState streams = 2;
+  }
+  map<string, TableCopyState> table_copy_state = 1;
+  map<string, ShardStreams> shard_streams = 2;
+}
+
+message WorkflowSwitchTrafficRequest {
+  string keyspace = 1;
+  string workflow = 2;
+  repeated string tablet_types = 3;
+  vttime.Duration max_replication_lag_allowed = 4;
+  bool enable_reverse_replication = 5;
+  int32 direction = 6;
```

Comment:
should we make an enum here (`WorkflowDirection { FORWARD = 0; BACKWARD = 1; }`) rather than relying on implied "0 is one way and 1 is the other"?